pkg/kubectl/cmd/autoscale.go

```diff
@@ -0,0 +1,164 @@
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	autoscaleLong = `Automatically increase or decrease number of pods deployed within the system as needed.
+
+<add more description>.`
+
+	autoscaleExample = `  # Auto scale a replication controller "foo":
+  $ kubectl autoscale rc foo --min=3 --max=10
+
+<add more examples>`
+)
+
+// TODO: support autoscale for deployments
+func NewCmdAutoscale(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	filenames := []string{}
+	cmd := &cobra.Command{
+		Use:     "autoscale (-f FILENAME | TYPE NAME | TYPE/NAME)",
+		Short:   "Auto-scale a replication controller on the server",
+		Long:    autoscaleLong,
+		Example: autoscaleExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunAutoscale(f, out, cmd, args, filenames)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().String("generator", "horizontalpodautoscaler/v1beta1", "The name of the API generator to use. Currently there is only 1 generator.")
+	cmd.Flags().Int("min", 1, "The lower limit for the number of pods that can be set by the autoscaler.")
+	cmd.Flags().Int("max", 0, "The upper limit for the number of pods that can be set by the autoscaler. It cannot be smaller than min. Required.")
+	cmd.MarkFlagRequired("max")
+	cmd.Flags().Int("cpu%", kubectl.DefaultCPUUtil, fmt.Sprintf("The target average CPU utilization (represented as a percent of requested CPU) over all the pods. If not specified it defaults to the target CPU utilization at %v%% of the requested resources.", kubectl.DefaultCPUUtil))
+	cmd.Flags().String("name", "", "The name for the newly created object.")
+	cmd.Flags().Bool("dry-run", false, "If true, only print the object that would be sent, without creating it.")
+	return cmd
+}
+
+func RunAutoscale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, filenames []string) error {
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	// validate flags
+	errs := []error{}
+	max, min, cpu := cmdutil.GetFlagInt(cmd, "max"), cmdutil.GetFlagInt(cmd, "min"), cmdutil.GetFlagInt(cmd, "cpu%")
+	switch {
+	case max < min:
+		errs = append(errs, fmt.Errorf("--max=COUNT is required, and COUNT must be greater than or equal to --min=%v", min))
+		fallthrough
+	case min < 0:
+		errs = append(errs, fmt.Errorf("--min=COUNT must be non-negative"))
+		fallthrough
+	case cpu > 100 || cpu < 0:
+		errs = append(errs, fmt.Errorf("CPU utilization (%%) should be between 0 to 100"))
+	}
+	return errors.NewAggregate(errs)
+
+	mapper, typer := f.Object()
+	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		ContinueOnError().
+		NamespaceParam(namespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, filenames...).
```

Comment:
That's a bug.

There should be something like:
kubectl.AddJsonFilenameFlag(cmd, &filenames, usage)